### PR TITLE
[Flax] Use `jax.jit` instead of `nnx.jit` + downgrade to Qwix `0.1.2`

### PR DESCRIPTION
**Pull request by jrplatin**
# Description

We shouldn't be using `nnx.jit` as it is not recommended for vLLM (according to @cgarciae ).  In addition, I temporarily downgrade to Qwix 0.1.2 as 0.1.5 showed 10x slower pre-compilation times.  Tracking bug from @jshin1394 here:

https://b.corp.google.com/issues/485996206

# Tests

Verified Qwen3-32B + Qwix perf on the 1k/4k set:

```
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Benchmark duration (s):                  856.68    
Total input tokens:                      1022406   
Total generated tokens:                  4096000   
Request throughput (req/s):              1.17      
Output token throughput (tok/s):         4781.22   
Peak output token throughput (tok/s):    6400.00   
Peak concurrent requests:                1000.00   
Total token throughput (tok/s):          5974.67   
---------------Time to First Token----------------
Mean TTFT (ms):                          326105.35 
Median TTFT (ms):                        237335.38 
P99 TTFT (ms):                           667220.75 
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          51.28     
Median TPOT (ms):                        52.41     
P99 TPOT (ms):                           55.59     
---------------Inter-token Latency----------------
Mean ITL (ms):                           51.28     
Median ITL (ms):                         41.77     
P99 ITL (ms):                            229.08    
==================================================
```

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Comment by cgarciae** [comment]
Just to clarify, `nnx.jit` is not depracated but for vLLM I'd recommend using `jax.jit` to avoid some python overhead.

**Review by patemotter — APPROVED** [review]

**Comment by jshin1394** [comment]
Can we confirm quantization correctness with the move to `jax.jit`? Given that `nnx.jit` is specifically designed to handle the lifting and merging of `nnx.Module` state, it seems safer for the initial model transformation. We could potentially use `nnx.jit` as a one-time cost inside `qwix_utils` and use the faster `jax.jit` path for the inference loop. @cgarciae, does this alignment between setup and serving make sense?

**Comment by cgarciae** [comment]
I think its fine to use `nnx.jit` for initialization.

**Comment by jrplatin** [comment]
Verified accuracy and perf here on MMLU:

```
============ Serving Benchmark Result ============
Successful requests:                     14042     
Benchmark duration (s):                  156.91    
Total input tokens:                      3313819   
Total generated tokens:                  42126     
Request throughput (req/s):              89.49     
Output token throughput (tok/s):         268.48    
Total token throughput (tok/s):          21388.25  
---------------Time to First Token----------------
Mean TTFT (ms):                          88047.79  
Median TTFT (ms):                        92667.11  
P99 TTFT (ms):                           153787.03 
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          182.47    
Median TPOT (ms):                        183.07    
P99 TPOT (ms):                           188.77    
---------------Inter-token Latency----------------
Mean ITL (ms):                           184.10    
Median ITL (ms):                         183.10    
P99 ITL (ms):                            189.85    
----------------End-to-end Latency----------------
Mean E2EL (ms):                          88412.74  
Median E2EL (ms):                        93035.89  
P99 E2EL (ms):                           154158.13 
==================================================
Evaluating MMLU...
[nltk_data] Downloading package punkt to
[nltk_data]     /home/jacobplatin_google_com/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package punkt_tab to
[nltk_data]     /home/jacobplatin_google_com/nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!

Results

{'accuracy': 0.8158, 'gen_num': 14042}
```

Matching: https://b.corp.google.com/issues/483986244#comment3
